bpf/lib/icmp6.h

```diff
@@ -347,18 +364,25 @@ static __always_inline int icmp6_send_time_exceeded(struct __ctx_buff *ctx,
 static __always_inline int __icmp6_handle_ns(struct __ctx_buff *ctx, int nh_off)
 {
 	union v6addr target, router;
+	struct endpoint_info *ep;
 
 	if (ctx_load_bytes(ctx, nh_off + ICMP6_ND_TARGET_OFFSET, target.addr,
 			   sizeof(((struct ipv6hdr *)NULL)->saddr)) < 0)
 		return DROP_INVALID;
 
 	cilium_dbg(ctx, DBG_ICMP6_NS, target.p3, target.p4);
 
+	ep = __lookup_ip6_endpoint(&target);
+
 	BPF_V6(router, ROUTER_IP);
 	if (ipv6_addrcmp(&target, &router) == 0) {
 		union macaddr router_mac = NODE_MAC;
 
-		return send_icmp6_ndisc_adv(ctx, nh_off, &router_mac);
+		return send_icmp6_ndisc_adv(ctx, nh_off, &router_mac, true);
+	} else if (ep) {
+		union macaddr node_mac = NODE_MAC;
+
+		return send_icmp6_ndisc_adv(ctx, nh_off, &node_mac, false);
```

Comment:
```suggestion
		union macaddr node_MAC = NODE_MAC;
		
		return send_icmp6_ndisc_adv(ctx, nh_off, &node_mac, !ep);
```

Reply:
Woops, I'm sorry, I misread the conditions on this block and made a bad suggestion here :(.

What I had in mind was to condense the whole `if`/`else if` block. Reading again, it could be written like this:
```c
bool router_target = ipv6_addrcmp(&target, &router);

if (router_target || ep) {
	union macaddr node_mac = NODE_MAC;

	return send_icmp6_ndisc_adv(ctx, nh_off, &node_mac, router_target);
}
```

Otherwise, your initial version was fine, too. But my suggestion of passing `!ep` to `send_icmp6_ndisc_adv()` on the two branches was wrong (you did it just for the second one which at least is technically correct, but not what we should do). Apologies about that. Would you mind changing again, please? (Either this suggestion or your original version, as you prefer.)